cuda-toolfile.spec

```diff
@@ -33,10 +34,142 @@ cat << \EOF_TOOLFILE >%{i}/etc/scram.d/cuda.xml
     <environment name="LIBDIR"    default="$CUDA_BASE/lib64"/>
     <environment name="INCLUDE"   default="$CUDA_BASE/include"/>
   </client>
-  <flags CUDA_CFLAGS="-fPIC"/>
-  <flags CUDA_FLAGS="-gencode arch=compute_35,code=sm_35 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_61,code=sm_61"/>
+  <flags CUDA_FLAGS="-gencode arch=compute_61,code=sm_61"/>
+  <flags CUDA_LDFLAGS="-L$(CUDA_BASE)/lib64"/>
+  <flags CUDA_LDFLAGS="-lcudadevrt"/>
   <runtime name="PATH" value="$CUDA_BASE/bin" type="path"/>
 </tool>
 EOF_TOOLFILE
 
+cat << \EOF_TOOLFILE >%{i}/etc/scram.d/cuda-cublas.xml
+<tool name="cuda-cublas" version="@TOOL_VERSION@">
+  <info url="https://docs.nvidia.com/cuda/cublas/index.html"/>
+  <use name="cuda"/>
+  <lib name="cublas"/>
+  <lib name="cublas_device"/>
+  <client>
+    <environment name="CUDA_CUBLAS_BASE" default="@TOOL_ROOT@"/>
+    <environment name="LIBDIR"           default="$CUDA_CUBLAS_BASE/lib64/stubs"/>
+    <environment name="INCLUDE"          default="$CUDA_CUBLAS_BASE/include"/>
```

Comment:
@fwyzard , this is going to make libs under stub directory available at runtime. I would suggest to create a new tool 
```
<tool name="cuda-stubs-interface" version="@TOOL_VERSION@">
  <info url="https://docs.nvidia.com/cuda/index.html"/>
  <client>
    <environment name="CUDA_STUBS_INTERFACE_BASE" default="@TOOL_ROOT@"/>
    <environment name="LIBDIR"          default="$CUDA_STUBS_INTERFACE_BASE/lib64/stubs"/>
    <environment name="INCLUDE"         default="$CUDA_STUBS_INTERFACE_BASE/include"/>
  </client>
  <flags SKIP_TOOL_SYMLINKS="1"/>
</tool>
```
and make all other tools to use it e.g. 
```
<tool name="cuda-stubs" version="@TOOL_VERSION@">
  <use name="cuda-stubs-interface"/>
  <lib name="cuda"/>
</tool>
<tool name="cuda-cublas" version="@TOOL_VERSION@">
  <use name="cuda"/>
  <lib name="cublas"/>
  <lib name="cublas_device"/>
  <flags CUDA_LDFLAGS="-lcublas_device"/>
</tool>
```

Reply:
Actually, that is just a mistake... the individual libraries should not need to define LIBDIR or INCLUDE, they should pick them up from "cuda-stubs" and "cuda".